src/lib/core/latch.h

```diff
@@ -127,24 +144,24 @@ latch_lock_timeout(struct latch *l, ev_tstamp timeout)
 	if (timeout <= 0)
 		return 1;
 
-	rlist_add_tail_entry(&l->queue, fiber(), state);
-	bool was_cancellable = fiber_set_cancellable(false);
```

Comment:
Let's split removing `fiber_set_cancellable` and other changes in two commits.
Removing `fiber_set_cancellable` seems unrelated to fixing latch order.